GitLab CI/CD failed

## Description
Hey team!
I saw that you added support for GitLab (beta). I tried to run scan against dummy repo hosted on GitLab, but unfortunately it failed. I created a token with the appropriate role and permissions.

<!--
Briefly describe the problem you are having in a few paragraphs.
-->

## What did you expect to happen?

The scan has run successfully

## What happened instead?

The scan immediately failed

Out from gitlab ci/cd runner:
```
$ chain-bench scan --repository-url $CI_PROJECT_URL --access-token $CHAIN_BENCH_TOKEN -o results.json --template @/templates/gitlab_security_scanner.tpl
2022-10-26 11:31:13 INF 🚩	Fetch Starting
2022-10-26 11:31:14 ERR error in fetching repository data
2022-10-26 11:[31](https://gitlab.com/XYZ/security/dummy-repo/-/jobs/3229947555#L31):14 INF 🛢️	Fetching Repository Settings Finished
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0xbd97a5]
goroutine 1 [running]:
github.com/aquasecurity/chain-bench/internal/scm-clients/clients.FetchClientData({0x7ffd195cb03b, 0x1a}, {0x7ffd195caff6?, 0x1?}, {0x0, 0x0})
	/home/runner/work/chain-bench/chain-bench/internal/scm-clients/clients/clients.go:40 +0x1a5
github.com/aquasecurity/chain-bench/internal/commands.NewScanCommand.func1(0xc000252a00?, {0xe3962a?, 0x8?, 0x8?})
	/home/runner/work/chain-bench/chain-bench/internal/commands/scan.go:22 +0xcc
github.com/spf13/cobra.(*Command).execute(0xc000252a00, {0xc00024e800, 0x8, 0x8})
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:872 +0x694
github.com/spf13/cobra.(*Command).ExecuteC(0xc000252780)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:990 +0x3b4
github.com/spf13/cobra.(*Command).Execute(...)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:918
github.com/aquasecurity/chain-bench/internal/commands.Execute({0xf93cc8?, 0xc0000021a0?})
	/home/runner/work/chain-bench/chain-bench/internal/commands/root.go:21 +0x[32](https://gitlab.com/tidio/security/automated-aws-audit/-/jobs/3229947555#L32)
main.main()
	/home/runner/work/chain-bench/chain-bench/cmd/chain-bench/main.go:12 +0x27
Uploading artifacts for failed job
```

--- a/internal/scm-clients/clients/clients.go
+++ b/internal/scm-clients/clients/clients.go
@@ -73,10 +73,10 @@ func FetchClientData(accessToken string, repoUrl string, branch string) (*checkm
 	}, checksIds, nil
 }
 
-func getRepoInfo(repoUrl string) (scm string, org string, repo string, err error) {
-	u, err := url.Parse(repoUrl)
+func getRepoInfo(repoFullUrl string) (string, string, string, error) {
+	u, err := url.Parse(repoFullUrl)
 	if err != nil || u.Scheme == "" {
-		logger.Errorf(err, "error in parsing repoUrl %s", repoUrl)
+		logger.Errorf(err, "error in parsing repoUrl %s", repoFullUrl)
 		if err == nil {
 			err = errors.New("error in parsing the host")
 		}
@@ -85,9 +85,13 @@ func getRepoInfo(repoUrl string) (scm string, org string, repo string, err error
 
 	path := strings.Split(u.EscapedPath(), "/")
 	if len(path) < 3 {
-		return "", "", "", fmt.Errorf("missing org/repo in the repository url: %s", repoUrl)
+		return "", "", "", fmt.Errorf("missing org/repo in the repository url: %s", repoFullUrl)
 	}
-	return u.Host, path[1], path[2], nil
+	repo := path[len(path) - 1]
+	namespace := strings.Split(u.Path, repo)[0]
+	namespaceTrimed := namespace[1:(len(namespace) - 1)]
+
+	return u.Host, namespaceTrimed, repo, nil
 }
 
 func getClientAdapter(scmName string, accessToken string) (adapter.ClientAdapter, error) {

--- a/internal/scm-clients/clients/clients.go
+++ b/internal/scm-clients/clients/clients.go
@@ -89,9 +89,9 @@ func getRepoInfo(repoFullUrl string) (string, string, string, error) {
 	}
 	repo := path[len(path) - 1]
 	namespace := strings.Split(u.Path, repo)[0]
-	namespaceTrimed := namespace[1:(len(namespace) - 1)]
+	trimedNamespace := namespace[1:(len(namespace) - 1)]
 
-	return u.Host, namespaceTrimed, repo, nil
+	return u.Host, trimedNamespace, repo, nil
 }
 
 func getClientAdapter(scmName string, accessToken string) (adapter.ClientAdapter, error) {

--- a/internal/scm-clients/clients/clients_test.go
+++ b/internal/scm-clients/clients/clients_test.go
@@ -1,0 +1,68 @@
+package clients
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type RepoInfo struct {
+	BaseUrl   string
+	Namespace string
+	Project   string
+}
+
+func TestGetRepoInfo(t *testing.T) {
+
+	tests := []struct {
+		Name        string
+		RepoUrl     string
+		ExpectedErr error
+		Expected    RepoInfo
+	}{{
+		Name:        "missing url schema",
+		RepoUrl:     "gitlab.com/rootgroup/subgroup/secondsubgroup/test",
+		ExpectedErr: fmt.Errorf("error in parsing the host"),
+		Expected:    RepoInfo{BaseUrl: "", Namespace: "", Project: ""},
+	}, {
+		Name:        "invalid url",
+		RepoUrl:     "https://gitlab?com/rootgroup/subgroup/secondsubgroup/test",
+		ExpectedErr: fmt.Errorf("missing org/repo in the repository url: %s", "https://gitlab?com/rootgroup/subgroup/secondsubgroup/test"),
+		Expected:    RepoInfo{BaseUrl: "", Namespace: "", Project: ""},
+	}, {
+		Name:        "github repo",
+		RepoUrl:     "https://github.com/aquasecurity/chain-bench",
+		ExpectedErr: nil,
+		Expected:    RepoInfo{BaseUrl: "github.com", Namespace: "aquasecurity", Project: "chain-bench"},
+	}, {
+		Name:        "gitlab project under root group",
+		RepoUrl:     "https://gitlab.com/rootgroup/test",
+		ExpectedErr: nil,
+		Expected:    RepoInfo{BaseUrl: "gitlab.com", Namespace: "rootgroup", Project: "test"},
+	}, {
+		Name:        "gitlab project under sub group",
+		RepoUrl:     "https://gitlab.com/rootgroup/subgroup/secondsubgroup/test",
+		ExpectedErr: nil,
+		Expected:    RepoInfo{BaseUrl: "gitlab.com", Namespace: "rootgroup/subgroup/secondsubgroup", Project: "test"},
+	}}
+
+	for _, test := range tests {
+		test := test
+		t.Run(test.Name, func(t *testing.T) {
+			t.Parallel()
+
+			baseUrl, namespace, project, err := getRepoInfo(test.RepoUrl)
+
+			if test.ExpectedErr == nil {
+				assert.NoError(t, err)
+			} else {
+				assert.EqualError(t, err, test.ExpectedErr.Error())
+			}
+			assert.Equal(t, test.Expected.BaseUrl, baseUrl)
+			assert.Equal(t, test.Expected.Namespace, namespace)
+			assert.Equal(t, test.Expected.Project, project)
+		})
+	}
+
+}

